src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ComputableGraphStructure.java

```diff
@@ -0,0 +1,286 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.cachemanager;
+
+import org.broadinstitute.hellbender.utils.Utils;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * This class computes a number of useful structural properties for DAG specified by the
+ * immediate parents and immediate children of each node.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class ComputableGraphStructure {
```

Comment:
As we discussed, it looks like many of the quantities computed here are not used (the getters are never called).  Depending on how ICG is refactored, I think you could just make that class do the essential computations, rather than having a CGS as a field.